TypeCobol.LanguageServer/Completion Factory/CompletionFactoryHelpers.cs

```diff
@@ -224,5 +230,82 @@ public static CompletionItem CreateCompletionItemForIndex(Compiler.CodeElements.
             var display = string.Format("{0} (from {1})", index.Name, variable.Name);
             return new CompletionItem(display) {insertText = index.Name, kind = CompletionItemKind.Variable};
         }
+
+        public static Case GetTextCase(string tokenText)
+        {
+            if (string.IsNullOrEmpty(tokenText)) return Case.Lower;
+            // check if upper case
+            bool isUpper = true;
+            foreach (char c in tokenText)
+            {
+                if (char.IsLower(c))
+                {
+                    isUpper = false;
+                    break;
+                }
+            }
+
+            if (isUpper) return Case.Upper;
+            // check if Camel case; ex. CamlSample
+            bool isCamel = false;
+            int iUpperMem = -1, count = tokenText.Length;
+            for (int i = 0; i < count; i++)
+            {
+                if (i == 0)
+                {
+                    // first letter
+                    if (char.IsLower(tokenText[i])) break;
+                    isCamel = true;
+                }
+                else
+                {
+                    if (char.IsUpper(tokenText[i]))
+                    {
+                        if (iUpperMem == (i - 1))
+                        {
+                            // 2 upper case characters one behind the other
+                            isCamel = false;
+                            break;
+                        }
+                        iUpperMem = i;
+                    }
+                }
+            }
+
+            if (isCamel && iUpperMem == count - 1) isCamel = false; // last char is upper
+            return (isCamel) ? Case.Camel : Case.Lower;
+        }
+
+        public const string ParamInput = "INPUT";
+        public const string ParamOutput = "OUTPUT";
+        public const string ParamInOut = "IN-OUT";
+        
+        public static Dictionary<string, string> GetParamsWithCase(Case textCase)
```

Comment:
Minor optimization : this helper dictionary could be declared as a static property to avoid multiple instanciations.
In that case, no need to change the signature of the Add*ToCompletionItems methods in CompletionFactory.

Reply:
I think also, it is possible to declare a static array of Dictionary<string, string> indexed by Case enum members, thus the method GetParamsWithCase can be removed.

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/tree/6d2502c2ac02e718a765ac3e549933b59b6ebb69